az-core/src/main/java/azkaban/utils/Props.java

```diff
@@ -481,17 +482,11 @@ public List<String> getStringListFromCluster(final String key) {
    * Returns a list of strings with the sep as the separator of the value
    */
   public List<String> getStringList(final String key, final String sep) {
-    final String val = get(key);
-    if (val == null || val.trim().length() == 0) {
+    String val = get(key);
+    if (val == null || (val = val.trim()).length() == 0) {
       return Collections.emptyList();
     }
-
-    if (containsKey(key)) {
-      return Arrays.asList(val.split(sep));
-    } else {
-      throw new UndefinedPropertyException("Missing required property '" + key
-          + "'");
-    }
```

Comment:
if condition should be guaranteed to be true by lines 484 and 485 so deleting it.